### PR TITLE
Update cypher-compiler version & cypher CHANGES for M02

### PR DESCRIPTION
**Pull request by boggle**

### CODE REVIEW

File: community/cypher/cypher/CHANGES.txt [hunk 1]
```diff
@@ -2,6 +2,10 @@
 -------
 
 Cypher:
+o Removed support for Cypher version 2.0 and 2.1, as announced
+o Produces warning if a query cannot be solved using the specified planner
+o Switched to using IDP as the new default planner
```
Comment (pontusmelke) [inline community/cypher/cypher/CHANGES.txt]: is `IDP` a public facing name?

File: community/cypher/cypher/src/main/scala/org/neo4j/cypher/internal/PathImpl.scala [hunk 2]
```diff
@@ -19,18 +19,22 @@
...
+
+import scala.collection.JavaConverters._
+import scala.collection.mutable
 
 // TODO: This is only here for accomodating cypher-compiler-1.9. Do not touch, do not import, purge it with fire post 2.3
 case class PathImpl(pathEntities: PropertyContainer*)
   extends org.neo4j.graphdb.Path
   with Traversable[PropertyContainer]
-  with CypherArray {
+  with
```
Comment (pontusmelke) [inline community/cypher/cypher/src/main/scala/org/neo4j/cypher/internal/PathImpl.scala]: Why the newline and why do we need to change this?

File: community/cypher/cypher/src/test/scala/org/neo4j/cypher/CartesianProductNotificationAcceptanceTest.scala [hunk 3]
```diff
@@ -21,7 +21,6 @@ package org.neo4j.cypher
 
 import org.mockito.Matchers._
 import org.mockito.Mockito.{verify, _}
-import org.neo4j.cypher.internal.NormalMode
 import org.neo4j.cypher.internal.compatibility.{StringInfoLogger2_3, WrappedMonitors2_3}
 import org.neo4j.cypher.internal.compiler.v2_3._
 import org.neo4j.cypher.internal.compiler.v2_3.notification.CartesianProductNotification
```
Comment (pontusmelke) [inline community/cypher/cypher/src/test/scala/org/neo4j/cypher/CartesianProductNotificationAcceptanceTest.scala]: Not needed

File: community/cypher/cypher/src/main/scala/org/neo4j/cypher/internal/PathImpl.scala (hunk 2 above)
Comment (boggle) [inline community/cypher/cypher/src/main/scala/org/neo4j/cypher/internal/PathImpl.scala]: Newline is a glitch; change has to do with class duplication / dependency cleanup we did in 2.2.2

File: community/cypher/cypher/CHANGES.txt [hunk 4]
```diff
@@ -2,6 +2,10 @@
 -------
 
 Cypher:
+o Removed support for Cypher version 2.0 and 2.1, as announced
+o Produces warning if a query cannot be solved using the specified planner
```
Comment (Mats-SX) [inline community/cypher/cypher/CHANGES.txt]: Shouldn't we mention something like "added configurable option" for this?

**Pull request merged**